assets/app/scripts/directives/editRequestLimit.js

```diff
@@ -0,0 +1,194 @@
+"use strict";
+
+angular.module('openshiftConsole')
+  .directive('computeResource', function($filter) {
+    return {
+      restrict: 'E',
+      require: 'ngModel',
+      scope: {
+        label: '@',
+        type: '@',
+        description: '@',
+        defaultValue: '=',
+        limitRangeMin: '=',
+        limitRangeMax: '=',
+        maxLimitRequestRatio: '=',
+        // If this is a limit, the value of the corresponding request to
+        // validate limit >= request.
+        request: '='
+      },
+      templateUrl: 'views/_compute-resource.html',
+      link: function(scope, elem, attrs, ngModel) {
+        scope.id = _.uniqueId('compute-resource-');
+        switch (scope.type) {
+        case 'cpu':
+          // Default to millicores.
+          scope.unit = scope.unit || "m";
+          scope.units = [{
+            value: "m",
+            label: "millicores"
+          }, {
+            value: "",
+            label: "cores"
+          }];
+          break;
+        case 'memory':
+          // Default to MiB.
+          scope.unit = scope.unit || "Mi";
+          scope.units = [{
+            value: "M",
+            label: "MB"
+          }, {
+            value: "G",
+            label: "GB"
+          }, {
+            value: "Mi",
+            label: "MiB"
+          }, {
+            value: "Gi",
+            label: "GiB"
+          }];
+          break;
+        }
+
+        var usageValue = $filter('usageValue');
+        var amountAndUnit = $filter('amountAndUnit');
+
+        var validateLimitRange = function() {
+          var value, minValid = true, maxValid = true;
+          if (scope.amount) {
+            value = usageValue(scope.amount + scope.unit, scope.unit);
```

Comment:
trying to follow exactly where unit normalization is happening with the current value vs the min and max.  some comments might help here